File: Day47/NoOf1Bits.java
import java.util.*;

public class NoOf1Bits {
    public int hammingWeight(int n) {
        int count = 0;
        while (n != 0) {
            n = n & (n - 1);
            count++;
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        NoOf1Bits sol = new NoOf1Bits();
        System.out.println(sol.hammingWeight(n));
    }
}

File: Day47/NoOf1Bits.py
class Solution(object):
    def hammingWeight(self, n):
        count = 0
        while n:
            n &= n - 1
            count += 1
        return count

if __name__ == "__main__":
    n = int(input().strip())
    sol = Solution()
    print(sol.hammingWeight(n))
